### PR TITLE
Unify CDEPS gfs, cfsr, and gefs datm datamodes + Improve error checking in rt.sh (2388) + Add ability to read increment files on native cubed sphere grid (2304)

### LINKED ISSUE #64: Synchronize DATM in EMC/CDEPS with authoritative ESCOMP/CDEPS
Code management should maintain two-way synchronization between the authoritative ESCOMP/CDEPS and EMC/CDEPS forks

### PR DESCRIPTION
**Pull request by NickSzapiro-NOAA**
## Commit Queue Requirements:

- [x] Fill out all sections of this template.
- [x] All sub component pull requests have been reviewed by their code managers.
- [x] Run the full Intel+GNU RT suite (compared to current baselines) on either Hera/Derecho/Hercules
- [x] Commit 'test_changes.list' from previous step
---
## Description:

Code for gfs, cfsr, and gefs datm datamodes is very similar (just with different function names). This PR removes the duplicates. As gefs datm datamode is already in ESCOMP/CDEPS, gefs is retained here. Note that GFS, GEFS, CFSR, etc. input datasets are still handled via streams in the same way, just with datamode == 'GEFS'. Sunset gfs_hafs datamode no longer used.

These PRs have also been merged for commit queue (via PR to this branch on their behalf):
- https://github.com/ufs-community/ufs-weather-model/pull/2304
- https://github.com/ufs-community/ufs-weather-model/pull/2388

### Commit Message:

```
* UFSWM -  Unify CDEPS gfs, cfsr, and gefs datm datamodes. Add ability to read increment files on native cubed sphere grid.  Update rt scripts to fix checking the job exit status from Slurm and PBS schedulers.
  * FV3 - Add namelist parameter for cubed sphere increment read and update atmos_cubed_sphere hash for GFDL Atmospheric Cubed Sphere PR #[342](https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/342)
    * atmos_cubed_sphere - Add ability to read increments from files on native cubed sphere grid
  * CDEPS - Unify CDEPS gfs, cfsr, and gefs datm datamodes. Rename CDEPS-related FILENAME_BASE to FILEBASE_DATM in tests. Sunset gfs_hafs datamode.
```

### Priority:

* Normal

### UFSWM:

* Closes https://github.com/NOAA-EMC/CDEPS/issues/64

### Sub component Pull Requests:

* CDEPS: https://github.com/NOAA-EMC/CDEPS/pull/63
* FV3: NOAA-EMC/fv3atm#837
  *  atmos_cubed_sphere: NOAA-GFDL/GFDL_atmos_cubed_sphere#342

### UFSWM Blocking Dependencies:

* None

---
### Regression Test Changes (Please commit test_changes.list):

* No Baseline Changes.

### Input data Changes:

* None.

### Library Changes/Upgrades:

* No Updates
  
---

## Testing Log:
- RDHPCS
  - [x] Hera
  - [x] Orion
  - [x] Hercules
  - [x] Jet
  - [x] Gaea
  - [x] Derecho
- WCOSS2
  - [x] Dogwood/Cactus
  - [x] Acorn
- [x] CI
- [ ] opnReqTest (complete task if unnecessary)

### CODE REVIEW

**Comment by NickSzapiro-NOAA** [comment]
The following CDEPS DATM tests are b4b on Hercules
```
====START OF HERCULES REGRESSION TESTING LOG====

UFSWM hash used in testing:
4eff4f18dfd4ad15608b7cefc79d07f9285e5e28

Submodule hashes used in testing:
 37cbb7d6840ae7515a9a8f0dfd4d89461b3396d1 AQM (v0.2.0-37-g37cbb7d)
 be5d28fd1b60522e6fc98aefeead20e6aac3530b AQM/src/model/CMAQ (CMAQv5.2.1_07Feb2018-198-gbe5d28fd1)
 fb19a4478ed0682df7aa3b69e26b764212b6068c CDEPS-interface/CDEPS (heads/simplify_datm)
 2ffee5cd48e0c389bdf75c0d910b3f6d53263563 CICE-interface/CICE (CICE6.0.0-451-g2ffee5c)
 f6ff8f7c4d4cb6feabe3651b13204cf43fc948e3 CICE-interface/CICE/icepack (Icepack1.1.0-182-gf6ff8f7)
 f13e16e414e115e268b2dd300b665e628e5f2429 CMEPS-interface/CMEPS (cmeps_v0.4.1-2308-gf13e16e)
 cabd7753ae17f7bfcc6dad56daf10868aa51c3f4 CMakeModules (v1.0.0-28-gcabd775)
 0495c19204325401ccba2943f99e65ee9190f07d FV3 (heads/develop)
 1720f85e54765251f869756e67c93ef7acefac0d FV3/atmos_cubed_sphere (201912_public_release-402-g1720f85)
 0f8232724975c13289cad390c9a71fa2c6a9bff4 FV3/ccpp/framework (2024-07-11-dev)
 2a50cccd916cceafaf031f4cd14f2ecef277be8f FV3/ccpp/physics (EP4-840-g2a50cccd)
 74a0e098b2163425e4b5466c2dfcf8ae26d560a5 FV3/ccpp/physics/physics/Radiation/RRTMGP/rte-rrtmgp (v1.6)
 be0410ece28f2b5b9c089f8ca09ce0c80c79fe6c FV3/upp (upp_v10.2.0-191-gbe0410ec)
-1ba8270870947b583cd51bc72ff8960f4c1fb36e FV3/upp/sorc/libIFI.fd
-7476b8f2790a47d788f79cebfdbb551567ae7cf8 FV3/upp/sorc/ncep_post.fd/post_gtg.fd
 041422934cae1570f2f0e67239d5d89f11c6e1b7 GOCART (sdr_v2.1.2.6-119-g0414229)
 bcf7777bb037ae2feb2a8a8ac51aacb3511b52d9 HYCOM-interface/HYCOM (2.3.00-122-gbcf7777)
 4b8777eb08b065d8a95e5317b6270a7a67ccf93e MOM6-interface/MOM6 (dev/master/repository_split_2014.10.10-10053-g4b8777eb0)
 9423197f894112edfcb1502245f7d7b873d551f9 MOM6-interface/MOM6/pkg/CVMix-src (9423197)
 29e64d652786e1d076a05128c920f394202bfe10 MOM6-interface/MOM6/pkg/GSW-Fortran (29e64d6)
 3ac32f0db7a2a97d930f44fa5f060c983ff31ee8 NOAHMP-interface/noahmp (v3.7.1-436-g3ac32f0)
 7f548c795a348bbb0fe4967dd25692c79036dc73 WW3 (6.07.1-346-g7f548c79)
 fad2fe9f42f6b7f744b128b4a2a9433f91e4296f stochastic_physics (ufs-v2.0.0-219-gfad2fe9)

NOTES:
[Times](Memory) are at the end of each compile/test in format [MM:SS](Size).
The first time is for the full script (prep+run+finalize).
The second time is specifically for the run phase.
Times/Memory will be empty for failed tests.

BASELINE DIRECTORY: /work/noaa/epic/hercules/UFS-WM_RT/NEMSfv3gfs/develop-20240730
COMPARISON DIRECTORY: /work2/noaa/stmp/nszapiro/stmp/nszapiro/FV3_RT/rt_915680

RT.SH OPTIONS USED:
* (-a) - HPC PROJECT ACCOUNT: fv3-cpu
* (-l) - USE CONFIG FILE: my_rt.conf
* (-e) - USE ECFLOW

PASS -- COMPILE 'datm_cdeps_intel' [08:10, 07:34] ( 2 remarks )
PASS -- TEST 'datm_cdeps_control_cfsr_intel' [03:15, 02:12](1125 MB)
PASS -- TEST 'datm_cdeps_restart_cfsr_intel' [02:16, 01:21](1106 MB)
PASS -- TEST 'datm_cdeps_control_gefs_intel' [02:16, 02:06](1011 MB)
PASS -- TEST 'datm_cdeps_iau_gefs_intel' [03:16, 02:10](1019 MB)
PASS -- TEST 'datm_cdeps_stochy_gefs_intel' [03:16, 02:11](1022 MB)
PASS -- TEST 'datm_cdeps_ciceC_cfsr_intel' [03:16, 02:13](1164 MB)
PASS -- TEST 'datm_cdeps_bulk_cfsr_intel' [03:15, 02:14](1142 MB)
PASS -- TEST 'datm_cdeps_bulk_gefs_intel' [03:16, 02:07](1030 MB)
PASS -- TEST 'datm_cdeps_mx025_cfsr_intel' [05:49, 04:54](1171 MB)
PASS -- TEST 'datm_cdeps_mx025_gefs_intel' [05:47, 04:55](1158 MB)
PASS -- TEST 'datm_cdeps_multiple_files_cfsr_intel' [03:14, 02:10](1163 MB)
PASS -- TEST 'datm_cdeps_3072x1536_cfsr_intel' [04:17, 03:04](2457 MB)
PASS -- TEST 'datm_cdeps_gfs_intel' [04:16, 03:08](2389 MB)

PASS -- COMPILE 'datm_cdeps_debug_intel' [06:10, 05:44] ( 2 warnings 2 remarks )
PASS -- TEST 'datm_cdeps_debug_cfsr_intel' [06:17, 05:11](1071 MB)

PASS -- COMPILE 'datm_cdeps_faster_intel' [06:10, 05:19] ( 2 remarks )
PASS -- TEST 'datm_cdeps_control_cfsr_faster_intel' [03:17, 02:14](1156 MB)

SYNOPSIS:
Starting Date/Time: 20240806 14:55:32
Ending Date/Time: 20240806 15:10:55
Total Time: 00h:15m:31s
Compiles Completed: 3/3
Tests Completed: 15/15

NOTES:
A file 'test_changes.list' was generated but is empty.
If you are using this log as a pull request verification, please commit 'test_changes.list'.

Result: SUCCESS

====END OF HERCULES REGRESSION TESTING LOG====
```

File: tests/default_vars.sh
Comment (junwang-noaa) [inline tests/default_vars.sh]: remove this line?

**Review by junwang-noaa — DISMISSED** [review]

**Comment by NickSzapiro-NOAA** [comment]
@DavidNew-NOAA @DusanJovic-NOAA If ok with you, your PRs have been merged into this PR for the commit queue. All tests pass b4b on Hera. If you make any more edits, please remember to push those changes here as well. I've combined commit messages as well if you want to take a look. 

@FernandoAndrade-NOAA If this looks ready to you, can you arrange reviews? Jun had previously approved the CDEPS one.

**Comment by DavidNew-NOAA** [comment]
@NickSzapiro-NOAA Works for me. Thanks

**Comment by zach1221** [comment]
@NickSzapiro-NOAA can you sync up your branch(s) once again? Once the CDEPS sub-pr has an approval we'd like to begin testing here.

**Comment by NickSzapiro-NOAA** [comment]
@zach1221 I can do this in 2 hours. I believe you have write access. Please feel free sync with develop before then, if it helps

**Comment by NickSzapiro-NOAA** [comment]
I think @DavidNew-NOAA needs to sync with FV3 hash as well for the branch referenced in the .gitmodules

**Comment by BrianCurtis-NOAA** [comment]
I'm not sure that AutoRT failures should be adding files to the logs directory.

**Comment by zach1221** [comment]
Disk quota exceeded on Hercules. I'll clean up the directory and try again.

**Comment by zach1221** [comment]
> I'm not sure that AutoRT failures should be adding files to the logs directory.

Ok, I'll remove the feature. Disk space looks like it's still an issue on hercules also.

**Comment by NickSzapiro-NOAA** [comment]
Doesn't the FV3 branch need to be synced again?
```
url = https://github.com/DavidNew-NOAA/fv3atm
branch = feature/cubed_sphere_inc
 ```

**Comment by BrianCurtis-NOAA** [comment]
> Doesn't the FV3 branch need to be synced again?
> 
> ```
> url = https://github.com/DavidNew-NOAA/fv3atm
> branch = feature/cubed_sphere_inc
> ```

Yes, no tests should be running right now until that is fixed.

**Comment by DavidNew-NOAA** [comment]
@BrianCurtis-NOAA @NickSzapiro-NOAA Sorry, I'm on PTL until tomorrow. I won't have access to my laptop until tomorrow to update my branch.

**Comment by zach1221** [comment]
Testing is complete. We can start the merging process with cubed-sphere [#342 ](https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/342)

**Comment by zach1221** [comment]
@NickSzapiro-NOAA Please update .gitmodule urls and hashes.

https://github.com/NOAA-EMC/CDEPS/commit/1f9eaaa142c8b07ed6b788c9f44ea02cc86d0bae
https://github.com/NOAA-EMC/fv3atm/commit/70b306581b24ab5e6b21e0a2b16b95f3a980660c

**Review by jkbk2004 — APPROVED** [review]

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Pull request merged**
